### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by nogahaskal**

### CODE REVIEW

File: src/vip/vip.interface.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export interface IUser {
+  _id?: string;
+  userID: string;
+  role: ROLE;
+}
+
+export enum ROLE {
+  READ = 'READ',
+  WRITE = 'READ'
```
Comment (shakedmanes) [inline src/vip/vip.interface.ts]: Should be `WRITE = 'WRITE'`?

**Review by shakedmanes — CHANGES_REQUESTED** [review]
Nice code!

File: src/vip/vip.interface.ts (hunk 1 above)
Comment (nogahaskal) [reply src/vip/vip.interface.ts]: haha it should.

Comment (nogahaskal) [reply src/vip/vip.interface.ts]: fixed that.

**Review by shakedmanes — APPROVED** [review]

File: src/authentication/authentication.handler.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+        const vip = await VipManager.getVipByID(parsedUser.id);
+        if (!vip) throw new NotPermittedError();
+
+        const userWithRole = {
+            ...parsedUser,
+            role: vip.role
+        }
+
+        const token = jwt.sign(userWithRole, config.authentication.secret);
```
Comment (Shahar-Y) [inline src/authentication/authentication.handler.ts:24]: type

File: src/utils/errors/handler.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import * as express from 'express';
+import { ServerError, UserError } from './application';
+
+export function userErrorHandler(error: Error, req: express.Request, res: express.Response,
```
Comment (Shahar-Y) [inline src/utils/errors/handler.ts]: req is not used. use `_` instead

**Review by Shahar-Y — APPROVED** [review]
Looks good to me, but I'm not the expert.
@AllKinds, any opinions?

File: src/authentication/authentication.handler.ts (hunk 2 above)
Comment (nogahaskal) [reply src/authentication/authentication.handler.ts:24]: jsonwebtoken library does not offers a type

File: src/utils/errors/handler.ts (hunk 3 above)
Comment (nogahaskal) [reply src/utils/errors/handler.ts]: done

**Pull request merged**